Allows users to download reports on bedspace usage and occupancy

patrickjfl opened the pull request:
## Screenshots of UI changes

### Before

![Screenshot 2023-05-11 at 14 55 53](https://github.com/ministryofjustice/hmpps-temporary-accommodation-ui/assets/70749355/ddc7f917-b319-4652-a623-d234d2ee5830)

### After

![Screenshot 2023-05-11 at 14 25 13](https://github.com/ministryofjustice/hmpps-temporary-accommodation-ui/assets/70749355/46e4440d-0fbe-42c9-a973-6878ab906420)

# Release checklist

[Release process documentation](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/edit-v2/4247847062?draftShareId=a1c360ab-bd31-4db1-aae3-7cc002761de9).

## Pre-merge checklist

- [ ] Are any changes required to dependent services for this change to work?
  (eg. CAS API)
- [ ] Have they been released to production already?

## Post-merge checklist

- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to test
- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to preprod
- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to prod

<!-- Should a release fail at any step, you as the author should now lead the work to
fix it as soon as possible. You can monitor deployment failures in CircleCI
itself and application errors are found in
[Sentry](https://ministryofjustice.sentry.io/issues/?project=4504129156218880&referrer=sidebar&statsPeriod=24h).
Both events should be automatically sent to our [Slack
channel](https://mojdt.slack.com/archives/C048BJS7S2F). -->

Commit:
Rename ‘booking report’ to ‘report’

This is in preparation for the booking report page becoming a page for multiple report types.

Commit:
Add ReportType type and getApiReportPath util

Adds a new type to reflect the possible report types, and a util for use in our reportClient to return the API path for a given report type.

Also updates the reportForProbationRegionFilename function to include the different report types in the generated file name.

Commit:
Updates report service to accommodate different report types

Commit:
Remove ‘bookings’ from reports path

As in the previous commit, now that the reports page is more generic, its paths are updated to reflect this.

Commit:
Updates integration tests

Commit:
Updates E2E tests

richpjames commented on server/@types/ui/index.d.ts:
```diff
@@ -269,3 +269,5 @@ export interface SideNavObj {
 }
 
 export type BookingSearchApiStatus = 'provisional' | 'confirmed' | 'arrived' | 'departed' | 'closed'
+
```
I think this change (updating the types) should be in the previous commit

patrickjfl replied:
oh yeah, good spot!

richpjames commented on server/controllers/temporary-accommodation/manage/reportsController.ts:
```diff
@@ -63,7 +63,7 @@ export default class ReportsController {
         )
       } catch (err) {
-        catchValidationErrorOrPropogate(req, res, err, paths.reports.bookings.new({}))
+        catchValidationErrorOrPropogate(req, res, err, paths.reports.new({}))
```
I think this change should be in one of the earlier commits

patrickjfl replied:
yes, thank you

richpjames commented on integration_tests/tests/temporary-accommodation/manage/report.cy.ts:
```diff
@@ -73,25 +75,126 @@ context('Report', () => {
+      // Then I should see the user's probation region preselected
+      page.shouldPreselectProbationRegion(this.actingUserProbationRegion)
+
+      // And when I fill out the form
```
Might just be me but I'm not sure about this 'and when'  
I think we should have another given, when, then. 
Maybe something like 'Given I need a report, when I fill out...'

patrickjfl replied:
yeah I agree. I've tidied it up a bit

richpjames reviewed (approved):
Looks good to me, just a couple of minor comments

Commit:
Update report client

Updates the reportForProbationRegion method on the ReportClient to fetch reports of different types.

Commit:
Updates reports controller to accommodate reports of different types
